### PR TITLE
JsonSchema generated classes are persisted

### PR DESCRIPTION
**Pull request by manusa**
## Description

Partially persist generated model in `src/generated/java/`.

This PR contains:
- Changed `pom.xml`:
  - Modified to allow an extra source directory
  - Configured additional execution to generate JsonSchema classes into that directory
- Modified generation scripts to include Maven build step to generate Java classes in addition to json schemas.

When reviewing you should consider the changed poms, shell scripts and CI workflow configurations (basically ignore any *.java file).

Relates to #2118 and #2119

## What does this PR bring?
- :+1: Lowered the compilation time (without tests) from ~7 to ~6 minutes ---> ~15%
- :+1: Should improve IDE experience, running a test from the IDE should now be possible and hassle free (still not perfect, but a considerable improvement should be noted)
- :-1: Model generation `./kubernetes-model-generator/generateModelDocker.sh` takes much longer since it involves a Maven execution for each module. This shouldn't really be a problem because it has only to be executed in case there are changes in our go files.
- :-1: + :+1: Similar to the previous point, the CI workflow "Go Build / Kubernetes Model Generator Build " takes longer to execute. Since this happens in the CI, it shouldn't be a problem either. On the other hand, we are now also verifying that generated Java classes won't change, which will allow us to detect early any trouble with the jsonschema2pojo code generation.

## Feedback
So far my IDE experience with the partially persisted model has improved.
It would be nice if those of you who had previous issues with the setup (@metacosm. @Fabian-K, @jorsol...) could provide your feedback to see if we should proceed and merge this or consider it a failed experiment. 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: kubernetes-model-generator/generateModelDocker.sh [hunk 1]
```diff
@@ -18,13 +18,13 @@
 set -e
 
 BASEDIR=$(dirname "$BASH_SOURCE")
-ABSOLUTE_BASEDIR=$(realpath "$BASEDIR")
+ABSOLUTE_BASEDIR=$(realpath "$BASEDIR/..")
 
-DOCKER_IMAGE_GOLANG=golang:1.15.6
+DOCKER_IMAGE_GOLANG=marcnuri/golang-1.15-java11
```
Comment (metacosm) [inline kubernetes-model-generator/generateModelDocker.sh:23]: Oooh, that looks like a maintenance nightmare!

Comment (manusa) [reply kubernetes-model-generator/generateModelDocker.sh:23]: :)

https://github.com/manusa/docker-images/blob/2dab5af4baef3616cc7ad02ba7c8d27addb32aa9/golang-1.15-java11/Dockerfile#L8

Not really, it's a very simple image, but I haven't found any other, open, publicly maintained equivalent (I bet not many Golang developers want to mix it with Java, and vice-versa)

**Review by oscerd — APPROVED** [review]

**Review by rohanKanojia — APPROVED** [review]
+1, I think it's a step forward to making the development experience better.

**Comment by Fabian-K** [comment]
Looking good! What are the plans for `/extensions`? I assume it makes sense to apply this there as well :)

**Comment by manusa** [comment]
> Looking good! What are the plans for `/extensions`? I assume it makes sense to apply this there as well :)

:) Yes, indeed. I'm just trying to make sure that we are on the right path before investing more time.

**Review by metacosm — APPROVED** [review]

**Pull request merged**